demos/fleet_provisioning/fleet_provisioning_with_csr/fleet_provisioning_payload_operations.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * AWS IoT Device SDK for Embedded C 202103.00
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/* Standard includes. */
+#include <stdlib.h>
+#include <stdbool.h>
+
+/**
+ * @brief Creates the document to be published to the CreateCertificateFromCsr
+ * API in order to request a certificate from AWS IoT for the included
+ * Certificate Signing Request (CSR).
+ *
+ * @param[in] pBuffer Buffer into which to write the publish document.
+ * @param[in] bufferLength Length of #buffer.
+ * @param[in] pCsr The CSR to include in the document.
+ * @param[in] csrLength The length of #pCsr.
+ * @param[out] pOutLengthWritten The length of the publish document.
+ */
+bool generateCsrRequest( char * pBuffer,
```

Comment:
As the payload is populated in in binary format, use of `char *` type is not semantically matching. It would be better to use `uint8_t *`.